airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -293,7 +312,30 @@ def read(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
 
         yield from self._filter_dynamic_fields(self._filter_old_records(self._transform(self._read(getter, params))))
 
+    @staticmethod
+    def _get_field_props(field_type: str) -> Mapping[str, str]:
+
+        if field_type in VALID_JSON_SCHEMA_TYPES:
+            return {
+                "type": field_type,
+            }
+
+        converted_type, field_format = KNOWN_CONVERTIBLE_SCHEMA_TYPES.get(field_type) or (None, None)
+
+        if not converted_type:
```

Comment:
let's default to string if type is unknown to avoid failures
```suggestion
        if not converted_type:
            converted_type = "string"
```

Reply:
@sherifnada in case we return `converted_type = "string"` in the schema, we should validate and convert the data at the transform step for each record, don't we? Meaning it can be an irreversible transformation for those fields which requires extra steps with no guarantee it works for every case. 

Can we just return `converted_type = "string"` for unknown fields without actual validation and converting for them?

@keu what do you think?

Reply:
@gaart my understanding is that without doing something like the suggestion above, the output schema will contain a field without any `type` field because of the `(None, None)` statement above. 

If we need to cast to a string that's acceptable IMO. alternatively we can skip the field altogether

Reply:
casting to string seems safe enough for unknown data types, which should be rare, we probably want some warning as well